src/osx/cocoa/textctrl.mm

```diff
@@ -562,6 +583,29 @@ - (void)cut:(id)sender
 
 - (void)paste:(id)sender
 {
+    NSPasteboard *pb = [NSPasteboard generalPasteboard];
```

Comment:
Are you sure this code is needed, i.e. `shouldChangeTextInRange:replacementString:` is really not invoked when pasting? I'd expect it to be...